app/Phragile/ActionHandler/SprintStoreActionHandler.php

```diff
@@ -0,0 +1,130 @@
+<?php
+namespace Phragile\ActionHandler;
+
+use Phragile\PhabricatorAPI;
+use \Flash;
+use Illuminate\Support\Facades\Redirect;
+
+class SprintStoreActionHandler {
+	private $userPhabricatorAPI;
+	private $botPhabricatorAPI;
+	private $redirect;
+	private $sprint;
+	private $user;
```

Comment:
Be careful when using fields for internal state. Having fields set in the constructor are fine. Having fields set at the start of the only public method is also fine. Anything else requires caution with regard to state bugs. Following these rules, I can see `userPhabricatorAPI` and `botPhabricatorAPI` are fine. Since `sprint` and `user` are set in `performAction`, they are probably fine, since the two other public methods don't appear to interact with them. That leaves `redirect`. And indeed, it does look like there is potential for a state bug. If the service is used once, it will keep the value for `redirect`, and not reset it if it is used again.